scipy/spatial/_hausdorff.pyx

```diff
@@ -34,8 +36,12 @@ def directed_hausdorff(double[:,::1] ar1, double[:,::1] ar2, seed=0):
     rng = np.random.RandomState(seed)
     resort1 = np.arange(N1)
     resort2 = np.arange(N2)
-    rng.shuffle(resort1)
-    rng.shuffle(resort2)
+    with warnings.catch_warnings():
+        warnings.filterwarnings("ignore",
+                                message="`x` isn't a recognized object")
+        rng.shuffle(resort1)
+        rng.shuffle(resort2)
```

Comment:
I'm just trying to understand--what's the danger of misbehavior here for data produced by `np.arange()`?

Or are you saying NumPy is likely going to change the behavior again (you mentioned special casing memoryviews) so this is a temporary work-around?

Reply:
This is a 1-D array, so should be safe. But the code path it's taking is in general not safe for possibly non-contiguous array-likes. Hence it started to trigger a new warning in 1.20